src/Core/Domain/Product/Command/UpdateProductCategoriesCommand.php

```diff
@@ -88,4 +87,32 @@ public function getCategoryIds(): array
     {
         return $this->categoryIds;
     }
+
+    /**
+     * @param int[] $categoryIds
+     */
+    private function setCategoryIds(array $categoryIds): void
+    {
+        $this->assertCategoryIdsAreNotEmpty($categoryIds);
+
+        $this->categoryIds = array_map(
+            function ($id) {
+                return new CategoryId($id);
+            }, $categoryIds
+        );
+    }
+
+    /**
+     * @param int[] $categoryIds
+     */
+    private function assertCategoryIdsAreNotEmpty(array $categoryIds): void
+    {
+        if (empty($categoryIds)) {
+            throw new LogicException(sprintf(
```

Comment:
I'm not a fan of generic exceptions, I prefer using domain exceptions because they can be catched independently. If you choose to keep a generic exception, you're choosing to let it bubble up. In that case, I think the more descriptive exception would be `InvalidArgumentException`.

Reply:
Indeed, I let it bubble up, because IMO it is purely technical error that does not involve any domain rules and the LogicException seems to fit this case.
According to PHP docs LogicException - `Exception that represents error in the program logic`.. Whereas InvalidArgumentException - `Exception thrown if an argument is not of the expected type.`

Reply:
Yep, as I said in my other comment I thought you were verifying each category id independently for some reason, my bad.

That said, this sounds more like a Runtime than Logic error because the size of the array cannot be inspected statically.